docs/csharp/language-reference/operators/switch-expression.md

```diff
@@ -1,54 +1,60 @@
 ---
 title: "switch expression - C# reference"
-description: Learn how to use the C# switch expression for pattern matching and other data introspection
-ms.date: 01/14/2021
+description: Learn about the C# switch expression that provides switch-like semantics based on pattern matching
+ms.date: 04/16/2021
+f1_keywords:
+  - "switch_CSharpKeyword"
+helpviewer_keywords:
+  - "switch expression [C#]"
+  - "pattern matching [C#]"
 ---
 # switch expression (C# reference)
 
-This article covers the `switch` expression, introduced in C# 8.0. For information on the `switch` statement, see the article on the [`switch` statement](../keywords/switch.md) in the [statements](../keywords/index.md) section.
+Beginning with C# 8.0, you use the `switch` expression to evaluate a single expression from a list of candidate expressions based on a pattern match with an input expression. For information about the `switch` statement that supports `switch`-like semantics in a statement context, see the [`switch` statement](../keywords/switch.md) article.
 
-## Basic example
-
-The `switch` expression provides for `switch`-like semantics in an expression context. It provides a concise syntax when the switch arms produce a value. The following example shows the structure of a switch expression. It translates values from an `enum` representing visual directions in an online map to the corresponding cardinal direction:
+The following example demonstrates a `switch` expression, which converts values of an [`enum`](../builtin-types/enum.md) representing visual directions in an online map to the corresponding cardinal directions:
 
 :::code language="csharp" source="snippets/shared/SwitchExpressions.cs" id="SnippetBasicStructure":::
 
-The preceding sample shows the basic elements of a switch expression:
+The preceding example shows the basic elements of a `switch` expression:
 
-- The *range expression*: The preceding example uses the variable `direction` as the range expression.
```

Comment:
That name, "range expression", has been always confusing to me. Why?

My guess, that comes from the [spec proposal](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/patterns#switch-expression). Particularly, from the following spec:
```csharp
switch_expression
    : range_expression 'switch' '{' '}'
    | range_expression 'switch' '{' switch_expression_arms ','? '}'
    ;
```

However, in the text, the proposal doesn't use "range expression" term but the following (emphasis mine):
> At runtime, the result of the switch_expression is the value of the expression of the first switch_expression_arm for which the expression on **the left-hand-side of the switch_expression** matches 

I think `range_expression` comes from the [System.Range spec](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges#systemrange) and is used to indicate the precedence of a `switch` expression. So, no need to use the same term in the docs. I think the "input expression" term is clearer.